addons/project/models/project.py

```diff
@@ -2219,7 +2219,25 @@ def _get_default_color(self):
 
     name = fields.Char('Name', required=True, translate=True)
     color = fields.Integer(string='Color', default=_get_default_color)
+    project_ids = fields.Many2many('project.project', 'project_project_project_tags_rel', string='Projects')
+    task_ids = fields.Many2many('project.task', string='Tasks')
 
     _sql_constraints = [
         ('name_uniq', 'unique (name)', "Tag name already exists!"),
     ]
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get_uid=None):
+        domain = [('name', operator, name)]
+        if 'project_id' in self.env.context:
+            project_id = self.env.context.get('project_id')
+            task_ids = self.env['project.task'].search([('project_id', '=', project_id)]).ids
+            domain = ['&', ('name', operator, name), '|', ('task_ids', 'in', task_ids), ('project_ids', 'in', project_id)]
+        return self._search(AND([args, domain]), limit=limit, access_rights_uid=name_get_uid)
```

Comment:
I saw in the parent method we check the `name` and the `operator` before adding the `('name', operator, name)`, perhaps we should do the same? 🤔 
Or we could call the parent method instead of calling `_search` method. 🙂 

```suggestion
        domain = args
        if 'project_id' in self.env.context:
            project_id = self.env.context.get('project_id')
            task_ids = self.env['project.task'].search([('project_id', '=', project_id)]).ids
            domain = AND([
                domain, 
                ['|', ('task_ids', 'in', task_ids), ('project_ids', 'in', project_id)]
            ])
        return super()._name_search(name, domain, operator, limit, name_get_uid)
```